x-pack/auditbeat/module/system/user/users_unix.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build !windows,cgo
+
+package user
+
+// #include <sys/types.h>
+// #include <pwd.h>
+import "C"
+
+import (
+	"github.com/pkg/errors"
+)
+
+// GetUsers retrieves a list of users using getpwent(3).
+func GetUsers() (users []*User, err error) {
+	C.setpwent()
+	defer C.endpwent()
+
+	for passwd, err := C.getpwent(); passwd != nil; passwd, err = C.getpwent() {
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting user")
+		}
+
+		// passwd is C.struct_passwd
+		users = append(users, &User{
+			name:     C.GoString(passwd.pw_name),
+			passwd:   C.GoString(passwd.pw_passwd),
```

Comment:
In older days this could contain the encrypted password (which we might indeed not want to send), but all modern UNIX systems set this to either `x` (password is in the shadow file) or `*` (login disabled). So I guess it depends on whether there is value in knowing whether the login for a user is disabled or not.

Reply:
I think it would be nice to translate them to some friendly values like `login_disabled` (*) or `in_shadow_file` (x) or `redacted` (all other values). Those would be more meaningful to me. WDYT?

Reply:
Makes sense, I've changed the possible values to `shadow_password`, `login_disabled`, `no_password` (which is a dangerous vulnerability since anybody can log in), and `<redacted>`.